lib/fast_jsonapi/serialization_core.rb

```diff
@@ -86,6 +86,24 @@ def record_hash(record, fieldset, includes_list, params = {})
         record_hash
       end
 
+      #
+      # It modifies cache options to prevent cache key collisions.
+      #
+      # If a fieldset is specified, it modifies the namespace to include the fields from the fieldset.
+      #
+      # If no fieldset is specified, the namespace will be unaltered.
+      #
+      # @param [Hash] options cache options hash
+      # @param [Array, nil] fieldset fieldset array or nil if unspecified
+      #
+      # @return [Hash] processed options hash
+      #
+      def process_cache_options(options, fieldset)
+        options = options ? options.dup : {}
+        options[:namespace] = "#{options[:namespace]}-fieldset:#{fieldset.join('_')}" if fieldset
```

Comment:
@wheatevo do you mind if we make this pretty please? :upside_down_face: 

First of all, if there's no namespace, we'll end up with something like `-fieldset:....<MODEL NAME>/id` [see cache key docs for more context](https://devdocs.io/rails~6.0/activerecord/integration#method-i-cache_key).

I'd like us to make sure that the namespace carries some context. Consider using the `jsonapi-<FIELDSETS VERSION>` if there's no namespace. If there's one available, let's just add the `-<FIELDSETS VERSION>` to it.

Another note here, is that I'm concerned about the size of this key. The caching stores seem to have a limit on the size of the final key, and it might be a better alternative to take care of it before we send it over to the caching store. Consider the approach [AS took on the memcache store](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/cache/mem_cache_store.rb#L162).

Let me know what you think?!

Reply:
Thank you for the quick review! I was not entirely happy with this solution so I appreciate your feedback.

I will make the requested changes and push up a new commit.

Reply:
Requested change has been made.

I ended up using SHA1 instead of MD5 to preserve FIPS compatibility. This should probably use `ActiveSupport::Digest` (https://github.com/rails/rails/pull/31289) if the active_support dependency is bumped in the future.